flink-runtime/src/main/java/org/apache/flink/runtime/concurrent/FutureUtils.java

```diff
@@ -1026,14 +1060,16 @@ public void onComplete(Throwable failure, U success) {
 	private static final class Timeout implements Runnable {
 
 		private final CompletableFuture<?> future;
+		private final String timeoutMsg;
 
-		private Timeout(CompletableFuture<?> future) {
+		private Timeout(CompletableFuture<?> future, @Nullable String timeoutMsg) {
 			this.future = checkNotNull(future);
+			this.timeoutMsg = timeoutMsg;
 		}
 
 		@Override
 		public void run() {
-			future.completeExceptionally(new TimeoutException());
+			future.completeExceptionally(null == timeoutMsg ? new TimeoutException() : new TimeoutException(timeoutMsg));
```

Comment:
this distinction isn't necessary; `new TimeoutException(timeoutMsg)` will work in any case.